perf/benchmark/security/generate_policies/generate_policies.go

```diff
@@ -80,180 +101,177 @@ func PolicyToYAML(policy *MyPolicy, spec proto.Message) (string, error) {
+	}
 
-	switch policy.Kind {
+	yaml, err := PolicyToYAML(policyHeader, spec)
+	if err != nil {
+		return "", err
+	}
+	return yaml, nil
+}
+
+func generateRules(policyData SecurityPolicy, policyHeader *MyPolicy) (string, error) {
+	switch policyHeader.Kind {
 	case "AuthorizationPolicy":
-		return generateAuthorizationPolicy(action, ruleToGenerator, policy, ruleMap)
+		return generateAuthorizationPolicy(policyData, policyHeader)
 	case "PeerAuthentication":
-		return "", fmt.Errorf("unimplemented")
+		return generatePeerAuthentication(policyData, policyHeader)
 	case "RequestAuthentication":
 		return "", fmt.Errorf("unimplemented")
 	default:
-		return "", fmt.Errorf("unknown policy kind: %s", policy.Kind)
+		return "", fmt.Errorf("unknown policy kind: %s", policyHeader.Kind)
 	}
 }
 
 func createPolicyHeader(namespace string, name string, kind string) *MyPolicy {
+	if namespace == "" {
+		namespace = "twopods-istio"
+	}
 	return &MyPolicy{
 		APIVersion: "security.istio.io/v1beta1",
 		Kind:       kind,
 		Metadata:   MetadataStruct{Namespace: namespace, Name: name},
 	}
 }
 
-func createRuleGeneratorMap(ruleToOccurancesPtr map[string]int) map[string]*ruleGenerator {
-	ruleGeneratorMap := make(map[string]*ruleGenerator)
-
-	if ruleToOccurancesPtr["numSourceIP"] > 0 || ruleToOccurancesPtr["numNamespaces"] > 0 {
-		ruleGeneratorMap["from"] = &ruleGenerator{
-			gen: sourceGenerator{},
-		}
-	}
+func generatePolicy(policyData SecurityPolicy, kind string, numPolicy int, totalPolicies int) (int, error) {
+	for i := 1; i <= numPolicy; i++ {
+		testName := fmt.Sprintf("test-%s-%d", kind, i)
+		policyHeader := createPolicyHeader(policyData.Namespace, testName, kind)
 
-	if ruleToOccurancesPtr["numPaths"] > 0 {
-		ruleGeneratorMap["to"] = &ruleGenerator{
-			gen: operationGenerator{},
+		rules, err := generateRules(policyData, policyHeader)
+		if err != nil {
+			return 0, err
 		}
-	}
-
-	if ruleToOccurancesPtr["numValues"] > 0 {
-		ruleGeneratorMap["when"] = &ruleGenerator{
-			gen: conditionGenerator{},
+		yaml := bytes.Buffer{}
+		yaml.WriteString(rules)
+		if totalPolicies > 1 {
+			yaml.WriteString("---")
 		}
+		totalPolicies--
+		fmt.Println(yaml.String())
 	}
-	return ruleGeneratorMap
+	return numPolicy, nil
 }
 
-func parseArguments(arguments string) (map[string]string, error) {
-	argumentMap := make(map[string]string)
-	// These are the default values
-	argumentMap["namespace"] = "twopods-istio"
-	argumentMap["policyType"] = "AuthorizationPolicy"
-	argumentMap["action"] = "DENY"
-	argumentMap["numPolicies"] = "1"
-
-	if len(arguments) > 0 {
-		for _, arg := range strings.Split(arguments, ",") {
-			keyValue := strings.Split(arg, ":")
-			if len(keyValue) == 1 {
-				return nil, fmt.Errorf("invalid argument: %s", keyValue[0])
-			}
-			argumentMap[keyValue[0]] = keyValue[1]
+func main() {
+	configFilePtr := flag.String("configFile", "", "The name of the config json file")
+	flag.Parse()
+
+	jsonString := ""
```

Comment:
why not make this `jsonBytes []byte` so that you don't need to convert it to string and then convert back?

Reply:
Good catch, will update